### ISSUE TITLE
Channel numbers should be available and match the EPG numbers.

### ISSUE THREAD

**Issue by tomidis** [Me-TV/Me-TV#16]
I think it is needed the possibility of channels order, so you can select the channel using the keyboard numbers, and also by remote controller which I have with my Terratec card, and I used it with Me-TV 1.3.7

**Comment by russel** [comment]
The issue of channel numbers turns out to be non-trivial. One could just assign integers to each channel which is I believe what Me TV 1.3.7 did. However these are not the DVB channel numbers which is what ought to be used, so that Me TV is behaving just as any other TV would – in particular the channel numbers of the EPG should be the channel numbers assigned for control. As far as I know just now, these channel numbers are not part of the tuning data, you have to process the MPEG-TS stream to get these channel numbers.  

Currently this work is blocked by not having a Rust-side binding for the MPEG-TS library. It would be possible to use the Rust C bindings but it would be far, far better to use the Rust-side bindings. Given there is the possibility of getting these in place next Saturday and Sunday, it would seem best to pause work on Me TV channel numbers till then.

**Comment by russel** [comment]
I have never tried using remote controllers. We should create a separate issue for remote controller functionality and leave this issue for  channel numbers in Me TV.

**Comment by russel** [comment]
Should be able to sort the channel list that appears on control buttons, the TV display frames, the full screen control bars, and the preferences dialog in channel number order where the channel numbers are those used in the EPG.

This raises the issue of whether this data should be stored between executions of Me TV. The channel files do not contain the EPG channel numbers, that is in the MPEG-TS data stream. The stream must be processed to get the data, and so it is not available immediately, and yet people will want the data available immediately. This will then raise the issue of checking consistency and handling changes to the channel numbers. It may not be necessary to do anything sophisticated (but it may be) since it may be that any change necessitates a retuning.

**Comment by russel** [comment]
Just to note that this issue depends on Issue #7 – though it doesn't need to be completed for this issue to be addressed, it is that there is a common infrastructure (handling the various tables accessed via MPEG-TS) that has to be ready.

**Comment by russel** [comment]
With GStreamer 1.16.0 now out and installable, work has started on being able to work with MPEG-TS from Me TV. See #7 for details.

**Comment by russel** [comment]
Me TV 3.1.x (3.1.0 has been released) now has the logical channel numbers for the channels. At least those that are sent out from the transmitter – I am finding that some channels found during dvbv5-scan do not seem to have their logical channel numbers published, weird and being investigated. The channel numbers are displayed with the channel names in the channel drop downs. Channels can be selected using the digits on the remote control. I guess using the numeric keys on the keyboard should be supported, but that is not the case just yet.

I am going to close this issue as completed. New issues should be raised if there any problems.
